scripts/upgrade_configuration.py

```diff
@@ -0,0 +1,184 @@
+args = parser.parse_args()
+
+if not args.add and not args.remove:
+    raise ValueError('Missing either add (-a or --add xxx) or remove arg (-r or --remove xxx)')
+if args.add and args.remove:
+    raise ValueError('Cannot set both add and remove args')
+if args.add and not args.value:
+    raise ValueError('Missing value (-v or --vvv xxx)')
+
+
+# pylint: disable=C0413
+from lbaf.IO.configurationValidator import ConfigurationValidator
+sections :dict = cast(dict, ConfigurationValidator.allowed_keys(group=True))
+from lbaf.Utils.logger import logger
+# pylint: enable=C0413
+
+logger = logger('upgrade', level='debug')
+logger.info('Script to bulk add or remove key to/from LBAF configuration files within the project')
+logger.info(
+    'NOTICE: Remember that the keys must be defined first at the schema level defined in ' \
+    'the ConfigurationValidator class'
+)
+
+def upgrade(file_path: Path) -> int:
+    """
+    Apply an upgrade to the given configuration file
+    """
+    print('---')
+    logger.debug('upgrading file %s ...', file_path)
+    key_path = None
+    if args.add:
+        logger.debug('Add key `%s` with value `%s`', args.add, args.value)
+        key_path = args.add.split('.')
+    elif args.remove:
+        print(f'Remove key `{args.remove}`')
+        key_path = args.remove.split('.')
+
+    parsed_value = args.value
+    if args.type is not None:
+        type_v = locate(args.type)
+        if callable(type_v):
+            parsed_value = type_v(args.value)
+
+    if key_path is None:
+        raise ValueError('The `key` must be a valid string')
+
+    conf = None
+    with open(file_path, 'r', encoding='utf-8') as yaml_file:
+        yaml_content = yaml_file.read()
+        conf =  yaml.safe_load(yaml_content)
+        node = conf
+        for i, key in enumerate(key_path):
+            is_leaf = i == len(key_path)-1
+            # if node does not exist create it
+            if not key in node.keys():
+                if args.add:
+                    if is_leaf:
+                        node[key] = parsed_value
+                    else:
+                        new_branch = {}
+                        new_node = new_branch
+                        new_key_path = key_path[i+1:]
+                        for j, new_key in enumerate(new_key_path):
+                            is_leaf = j == len(new_key_path) - 1
+                            new_node[new_key] = parsed_value if is_leaf else {}
+                            new_node = new_node[new_key]
+                        node[key] = new_branch
+            elif is_leaf:
+                if args.remove:
+                    del node[key]
+                else:
+                    node[key] = parsed_value
+            # go next node in child tree
+            if is_leaf:
+                break
+            else:
+                node=node[key]
+
+    def write_node(k, value, yaml_file):
+        """Wirtes a single node (key/value) in the given yaml file"""
```

Comment:
Typo (Wirtes/Writes); no s at end of very; missing space before W

Reply:
Thanks @ppebay it will be corrected